nirc_ehr/resources/queries/study/housing.js

```diff
@@ -4,19 +4,66 @@ var prevDate;
 
 EHR.Server.TriggerManager.registerHandlerForQuery(EHR.Server.TriggerManager.Events.BEFORE_INSERT, 'study', 'housing', function (helper, scriptErrors, row, oldRow) {
 
-    var isTransfer = prevAnimalId === row.Id;
+    if (helper.isETL()) {
+        var isTransfer = prevAnimalId === row.Id;
 
-    if (isTransfer) {
-        row.endDate = prevDate;
+        if (isTransfer) {
+            row.endDate = prevDate;
+        }
     }
 
-    if (row.endDate === 'undefined') {
-        console.log("end date not found for animal event - " + row.objectId);
+        if (row.endDate === 'undefined') {
+            console.log("end date not found for animal event - " + row.objectId);
+        }
+
+        prevAnimalId = row.Id;
+        prevDate = row.date;
+    });
+
+function onComplete(event, errors, helper){
+
+    // Similar to EHR housing trigger onComplete with minor differences to handle ETL
+    var idsToClose = [];
+    if (helper.isETL()) {
+        var boundaryRows = helper.getRows();
+        if (boundaryRows && boundaryRows.length > 0) {
+            boundaryRows = [boundaryRows[0]];  // Just need first boundary row for this extra step
```

Comment:
Why does this create a single element array and then do a for loop over it? Could we just use boundaryRows[0] directly and skip the array recreation and loop?

Reply:
I cleaned this up.  Now it is sharing that loop with the non-ETL case.